### PR TITLE
[Experimental] [CSharp Core] Fix World Translations

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Modified `ComponentTranslations` to match new Dispatcher interface.
Modified `SpatialOSReceiveSystem` to use new Dispatcher interface.

#### Tests
How did you test these changes prior to submitting this pull request?
What automated tests are included in this PR?
#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Assets/Gdk/Core/Components/WorldCommandsTranslation.cs [hunk 1]
```diff
@@ -111,14 +112,15 @@ public WorldCommandSender(long entityId, uint handleToTranslation)
             HandleToTranslation = handleToTranslation;
         }
 
-        public void SendCreateEntityRequest(Worker.Entity entity, long entityId = 0, uint timeoutMillis = 0)
+        public void SendCreateEntityRequest(Worker.Core.Entity entity, long? entityId, uint? timeoutMillis)
```
Comment (zeroZshadow) [inline workers/unity/Assets/Gdk/Core/Components/WorldCommandsTranslation.cs]: These should still have a default value of .. having no value?

File: workers/unity/Assets/Gdk/Core/Components/WorldCommandsTranslation.cs [hunk 2]
```diff
@@ -129,7 +131,7 @@ public void SendCreateEntityRequest(Worker.Entity entity, long entityId = 0, uin
             });
         }
 
-        public void SendDeleteEntityRequest(long entityId, uint timeoutMillis = 0)
+        public void SendDeleteEntityRequest(long entityId, uint? timeoutMillis)
```
Comment (zeroZshadow) [inline workers/unity/Assets/Gdk/Core/Components/WorldCommandsTranslation.cs]: default value

File: workers/unity/Assets/Gdk/Core/Components/WorldCommandsTranslation.cs [hunk 3]
```diff
@@ -141,7 +143,7 @@ public void SendDeleteEntityRequest(long entityId, uint timeoutMillis = 0)
             });
         }
 
-        public void SendReserveEntityIdsRequest(uint numberOfEntities, uint timeoutMillis = 0)
+        public void SendReserveEntityIdsRequest(uint numberOfEntities, uint? timeoutMillis)
```
Comment (zeroZshadow) [inline workers/unity/Assets/Gdk/Core/Components/WorldCommandsTranslation.cs]: default value

File: workers/unity/Assets/Gdk/Core/Components/WorldCommandsTranslation.cs [hunk 4]
```diff
@@ -153,7 +155,7 @@ public void SendReserveEntityIdsRequest(uint numberOfEntities, uint timeoutMilli
             });
         }
 
-        public void SendEntityQueryRequest(Worker.Query.EntityQuery entityQuery, uint timeoutMillis = 0)
+        public void SendEntityQueryRequest(Worker.Query.EntityQuery entityQuery, uint? timeoutMillis)
```
Comment (zeroZshadow) [inline workers/unity/Assets/Gdk/Core/Components/WorldCommandsTranslation.cs]: default value

File: workers/unity/Assets/Gdk/Core/Systems/SpatialOSReceiveSystem.cs [hunk 5]
```diff
@@ -55,15 +59,46 @@ private void OnDisconnect(DisconnectOp op)
...
             dispatcher.OnCriticalSection(op => { inCriticalSection = op.InCriticalSection; });
 
+            dispatcher.OnAddComponent(OnAddComponent);
+            dispatcher.OnRemoveComponent(OnRemoveComponent);
+            dispatcher.OnComponentUpdate(OnComponentUpdate);
+            dispatcher.OnAuthorityChange(OnAuthorityChange);
+
             foreach (var translationUnit in view.TranslationUnits.Values)
             {
+                if (translationUnit is IDispatcherCallbacks)
```
Comment (zeroZshadow) [inline workers/unity/Assets/Gdk/Core/Systems/SpatialOSReceiveSystem.cs]: This feels odd, dont all units have this interface?

Comment (jamiebrynes7) [reply workers/unity/Assets/Gdk/Core/Systems/SpatialOSReceiveSystem.cs]: WorldCommandsTranslation....

File: workers/unity/Assets/Gdk/Core/Components/ComponentTranslation.cs [hunk 6]
```diff
@@ -115,13 +93,37 @@ protected void RemoveComponents<T>(ref EntityCommandBuffer entityCommandBuffer,
                 }
             }
         }
+
+        protected static class TranslationErrors {
```
Comment (martinzlocha) [inline workers/unity/Assets/Gdk/Core/Components/ComponentTranslation.cs]: This no longer exists in master. Please rebase before merging.

Comment (jamiebrynes7) [reply workers/unity/Assets/Gdk/Core/Components/ComponentTranslation.cs]: This branch isn't targeting `master` 😉

**Comment by jamiebrynes7** [comment]
@zeroZshadow Going to close this as its gonna be superseded by a ton of other changes.

**Pull request closed without merging**